cephfs/cephfs.go

```diff
@@ -28,15 +37,50 @@ func CreateMount() (*MountInfo, error) {
 	if ret == 0 {
 		return mount, nil
 	} else {
+		log.Errorf("CreateMount: Failed to create mount")
 		return nil, CephError(ret)
 	}
 }
 
+func (mount *MountInfo) RemoveDir(path string ) error {
+	c_path := C.CString(path)
+	defer C.free(unsafe.Pointer(c_path))
+
+	ret := C.ceph_rmdir(mount.mount, c_path)
+	if ret == 0 {
+		return nil
+	} else {
+		log.Errorf("RemoveDir: Failed to remove directory")
```

Comment:
I am always in favor of verbose logging (hence the addition of logrus).  I dont think it harms the users unless they are worried about logging output, in which case they can set logrus level to Fatal.

Reply:
I'm not opposed to logging in principle in go-ceph, but there are two issues that I want to address:

1) we seem to be _maybe_ saying logrus is what we are going to use for logging---but there are a lot of logging libraries out there. will users face compatibility issues if we force this choice? on the logrus docs page, it suggests that libraries can use an abstraction that may fix this, allowing clients of reusable libraries to use the built-in logging _or_ logrus: https://gowalker.org/github.com/sirupsen/logrus#StdLogger

2) it isn't clear what value logging errors like EPERM from RemoveDir really offer since it truly _is not_ an actual error. It's only an error if the client decides it is an error--go-ceph is just reporting results--the client can log an application specific interpretation of the error if one exists. This is a super common thing that comes up with reusable libraries.

Reply:
>  we seem to be maybe saying logrus is what we are going to use for logging---but there are a lot of logging libraries out there. will users face compatibility issues if we force this choice? on the logrus docs page, it suggests that libraries can use an abstraction that may fix this, allowing clients of reusable libraries to use the built-in logging or logrus: https://gowalker.org/github.com/sirupsen/logrus#StdLogger

The standard logger (`import "log"`) isnt all that great in my opinion.  Having 3 modes: Print, Fatal, and Panic isnt very helpful for diversifying what the output is.  Many other languages support Debug, Info, Warning, Error, etc.  I dont want to shoe horn future developers to using logrus, if someone comes along and wishes to add a new and better logger, I would be all for a `sed 's/log\.Errorf/blah.Logger/g'`.  If it is easier, I can remove all of these print statements.  But I think it is useful for clients/developers to know which function is throwing the error without looking to a line number to correlate the error and the function.

>    it isn't clear what value logging errors like EPERM from RemoveDir really offer since it truly is not an actual error. It's only an error if the client decides it is an error--go-ceph is just reporting results--the client can log an application specific interpretation of the error if one exists. This is a super common thing that comes up with reusable libraries.

I didnt modify this portion of code, but prior to my modifications there was a function called `func (e CephError) Error() string `.  Every function called this function on an error.  This is a conversation that is a bit over my head, but all non-zero returns are by definition- errors.  I agree with you that if the client has the contextual knowledge to handle the error as either being malicious or benign.  All this code does is alert the user to the error, how they handle it is up to them.

Reply:
I'm ok merging the code as is--the logging isn't currently hurting anything, but I think the logging discussion is not over, and I wouldn't be surprised if its removal is requested later.

>but all non-zero returns are by definition- errors. 

I definitely disagree with this. It is super common for applications to do things like

```
int ret = stat("file", ...)
if (ret == -ENONET)
  file_exists = true
else ...
```

In this case, -ENOENT is not an error, it's just saying that the file doesn't exist. Other applications may treat this as an error, but the interface doesn't force one interpretation or the other.

Reply:
I am being too pedantic, so i'll defer the discussion to you.  I think what you are arguing is that it may not be a semantic error, while i am arguing that it is a literal error.

Reply:
What even is an error, anyway? :) All I'm trying to say is that some applications care and others  do not. Therefore, as a general purpose library, the most general solution is to not care.